Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx

```diff
@@ -104,10 +114,12 @@ function createItemsAndGroups(
     })
     .reduce(
       (result: { items: any[]; groups: IGroup[] }, dialog) => {
-        result.groups.push(createGroupItem(dialog, dialogId, position));
+        result.groups.push(createGroupItem(dialog, dialogId, position, warningContent));
         position += dialog.triggers.length;
+        const isRegEx = (dialog.content?.recognizer?.$kind ?? '') === regexRecognizerKey;
         dialog.triggers.forEach((item, index) => {
-          result.items.push(createItem(item, index));
+          const withWarning = isRegEx && (item.type === qnaMatcherKey || item.type === onChooseIntentKey);
```

Comment:
Take care of the logic of `isRegEx` and `withWarning`.
1. It's improper that ProjectTree only receives a constant string `warningContent` as a prop and this constant string is passed to several functions
2. Try to define a unified validator which validates each group item and generates warning messages